lib/commons/color/element-is-distinct.js

```diff
@@ -24,64 +11,105 @@ function _getFonts(style) {
  * @param  {HTMLElement} ancestorNode The ancestor node element to check
  * @return {Boolean}
  */
-function elementIsDistinct(node, ancestorNode) {
-  var nodeStyle = window.getComputedStyle(node);
+export default function elementIsDistinct(node, ancestorNode) {
+  const nodeStyle = window.getComputedStyle(node);
 
-  // Check if the link has a background
-  if (nodeStyle.getPropertyValue('background-image') !== 'none') {
-    return true;
-  }
+  const ancestorStyle = window.getComputedStyle(ancestorNode);
+  let curNode = node;
+  while (curNode !== ancestorNode) {
+    const curNodeStyle = window.getComputedStyle(curNode);
 
-  // Check if the link has a border or outline
-  var hasBorder = ['border-bottom', 'border-top', 'outline'].reduce(
-    (result, edge) => {
-      var borderClr = new Color();
-      borderClr.parseString(nodeStyle.getPropertyValue(edge + '-color'));
-
-      // Check if a border/outline was specified
-      return (
-        result ||
-        // or if the current border edge / outline
-        (nodeStyle.getPropertyValue(edge + '-style') !== 'none' &&
-          parseFloat(nodeStyle.getPropertyValue(edge + '-width')) > 0 &&
-          borderClr.alpha !== 0)
-      );
-    },
-    false
-  );
-
-  if (hasBorder) {
-    return true;
+    // Check if the link has a background
+    if (curNodeStyle.getPropertyValue('background-image') !== 'none') {
+      return true;
+    }
+
+    if (
+      hasSameText(node, curNode) &&
```

Comment:
I'm a little confused, why is hasSameText() not in the while condition? I.e. 
```js
while (curNode !== ancestorNode && hasSameText(node, curNode)) {
```

My intuition says that that's what should be happening. Remind me why an ancestor with a background-image that has different text would count as making the link distinct?